### PR TITLE
D8CORE-7055: added an aria-hidden is true

### PR DESCRIPTION
**Pull request by jenbreese**
# READY FOR REVIEW

# Summary
- Adding an `aria-hidden=true` to the block label on the secondary navigation.
- This will prevent a SR from reading it
- Pookmish: I added you in case there's another you'd prefer to do this. 

# Review By (Date)
- 11/13

# Criticality
- normal
- a11y fix

# Urgency
- normal

# Review Tasks

## Setup tasks and/or behavior to test

1. Check out this branch
2. Rebuild Cache and import config `drush cr ; drush ci`
3. Navigate to a page with secondary navigation. This doesn't effect the news, publications, people, etc list pages.
4. Verify that the `h2` that is hidden visiually now has the aria-hidden=true.
5. example:
Screenshot 2023-11-09 at 1 01 54 PM
6. Try it out with a SR if you'd like. Maybe we can get @cjwest to look at a Gitpod to verify it.

### Site Configuration Sync

- Is there a config:export in this PR that changes the config sync directory? NO

## Front End Validation
- [ ] Design is approved by @ user?
- [ ] HTML validation: Is the markup using the appropriate semantic tags and [passes validation](https://validator.w3.org/nu/)? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?
- [ ] Cross-browser testing: Has been performed? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?
- [ ] Automated accessibility: Scans performed? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?
- [ ] Manual accessibility: Manually tested? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?

### Code
- [ ] Are the naming conventions following our standards?
- [ ] Does the code have sufficient inline comments?
- [ ] Is there anything in this code that would be hidden or hard to discover through the UI?
- [ ] Are there any [code smells](https://blog.codinghorror.com/code-smells/)?
- [ ] Are tests provided? eg (unit, behat, or codeception)

### Code security
- [ ] Are all [forms properly sanitized](https://www.drupal.org/docs/8/security/drupal-8-sanitizing-output)?
- [ ] Any obvious [security flaws or new areas for attack](https://www.drupal.org/docs/8/security)?

## General
- [ ] Is there anything included in this PR that is not related to the problem it is trying to solve?
- [ ] Is the approach to the problem appropriate?

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
[- D8CORE-7055](https://stanfordits.atlassian.net/browse/D8CORE-7055)

# Resources
- [AMP Tool](https://stanford.levelaccess.net/index.php)
- [Accessibility Manual Test Script](https://docs.google.com/document/d/1ZXJ9RIUNXsS674ow9j3qJ2g1OAkCjmqMXl0Gs8XHEPQ/edit?usp=sharing)
- [HTML Validator](https://validator.w3.org/)
- [Browserstack](https://live.browserstack.com/dashboard) and link to [Browserstack Credentials](https://asconfluence.stanford.edu/confluence/display/SWS/External+Account+Credentials)

### CODE REVIEW

**Review by pookmish — COMMENTED** [review]
This should be done on the template layer  or config instead of javascript.

**Comment by jenbreese** [comment]
I think this can be improved. But is this right track?

File: themes/stanford_basic/templates/block/block--system-menu-block.html.twig [hunk 1]
```diff
@@ -45,8 +45,9 @@
   {% if not configuration.label_display %}
     {% set title_attributes = title_attributes.addClass('visually-hidden') %}
   {% endif %}
+
   {{ title_prefix }}
-  <h2{{ title_attributes.setAttribute('id', heading_id) }}>{{ configuration.label }}</h2>
+  <h2{{ title_attributes.setAttribute('id', heading_id) }} {% if not configuration.label_display %}aria-hidden=true{% endif %}>{{ configuration.label }}</h2>
```
Comment (pookmish) [inline themes/stanford_basic/templates/block/block--system-menu-block.html.twig]: I'd like to put this in a preprocess hook that applies to all blocks on every page, not just this one.

Comment (jenbreese) [reply themes/stanford_basic/templates/block/block--system-menu-block.html.twig:50]: Ok.  How about this @pookmish?

File: themes/stanford_basic/stanford_basic.theme [hunk 2]
```diff
@@ -226,6 +226,10 @@ function stanford_basic_preprocess_block(&$variables) {
       unset($variables['content']);
     }
   }
+
+  if (!isset($variables['configuration']['label_display'])) {
+    $variables['attributes']['aria-hidden'] = 'true';
```
Comment (jenbreese) [inline themes/stanford_basic/stanford_basic.theme]: How do I get it applied to the `h2` instead? I haven't figured that part out.  Also is this what you were thinking?

Comment (pookmish) [reply themes/stanford_basic/stanford_basic.theme]: By looking at the template, you can see that it uses the `title_attributes` variable: https://github.com/SU-SWS/stanford_profile/blob/8a78aba443667c54ccf3189211b98079f6345c4b/themes/stanford_basic/templates/block/block--system-menu-block.html.twig#L50

**Pull request merged**